### PR TITLE
Pre parse qc performance

### PR DESCRIPTION
**Pull request by eschechter**

### CODE REVIEW

**Comment by ohsqueezy** [comment]
Looks good, and we should use the optimization either way, but does explicitly closing the dataset and/or using `del` or even changing xarray versions help with this issue

File: gridded_etl_tools/utils/zarr_methods.py [hunk 1]
```diff
@@ -1086,38 +1086,46 @@ def pre_parse_quality_check(self, dataset: xr.Dataset):
...
-                unit = dataset[self.data_var()].encoding["units"]
                 if unit in self.extreme_values_by_unit.keys():
                     limit_vals = self.extreme_values_by_unit[unit]
                     if not limit_vals[0] <= random_val <= limit_vals[1]:
                         raise ValueError(
                             f"Value {random_val} falls outside acceptable range\
-                            {limit_vals} for data in units {unit}. Found at {random_coords}"
+                            {limit_vals} for data in units {unit}. Found at {current_coords}"
                         )
             # Build a dictionary of checked values to compare against after parsing
```
Comment (rbanick) [inline gridded_etl_tools/utils/zarr_methods.py:1128]: Should ditch this comment as well

**Review by rbanick — APPROVED** [review]

File: gridded_etl_tools/utils/zarr_methods.py [hunk 2]
```diff
@@ -1411,7 +1440,8 @@ def check_value(
             orig_val = orig_ds.sel(**selection_coords, method="nearest")[self.data_var()].values
         else:
             orig_val = orig_ds[self.data_var()].sel(**selection_coords).values
-        prod_val = prod_ds[self.data_var()].sel(**selection_coords).values
+        prod_selection_coords = selection_coords if prod_ds_z_index is None else {"z": prod_ds_z_index}
```
Comment (rbanick) [inline gridded_etl_tools/utils/zarr_methods.py:1444]: when would the `prod_ds_z_index` be `None`?

**Review by rbanick — APPROVED** [review]

**Comment by rbanick** [comment]
Superseded by https://github.com/Arbol-Project/gridded-etl-tools/pull/131

**Pull request closed without merging**